woocommerce/woocommerce
# [Accessibility] Add screen reader text to the gallery trigger link and update it to a…

… button

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #43574 .
- updates gallery trigger to a button
- adds screen reader text announcing what the button does (view gallery and has a popup)
- since the trigger has been changed to a button (was a link), themes like Storefront, will need to adjust their styling



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. View a single product
2. Using a screen reader, tab to the Gallery Trigger button (magnification glass in upper right corner of main product image)
3. It should announce that the button is a popup and will "View full-screen image gallery"
![image](https://github.com/user-attachments/assets/2351e388-023e-4ccc-98c2-4a403b6fe600)




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
